userspace/libsinsp/parsers.cpp

```diff
@@ -2035,6 +2036,10 @@ void sinsp_parser::parse_open_openat_creat_exit(sinsp_evt *evt)
 		ASSERT(parinfo->m_len == sizeof(uint32_t));
 		flags = *(uint32_t *)parinfo->m_val;
 
+		parinfo = evt->get_param(4);
```

Comment:
Please take a look at https://github.com/draios/sysdig/wiki/Modify-a-pre-existent-event-or-block-entry and patch the changes to this function to safely access event parameters that may be not available when reading older captures.

Reply:
I'd like to add the device field to scap files as well. Am I correct that this will require introducing SCAP_FILE_V3?

Reply:
No, as for updates of pre-existent events, changing the block entry version is no longer required. In the same document I've linked to you above there's a section describing the high level changes that you'll need to make.
To test the result, you can create captures with and without the new field and verify (with a debug-enabled sysdig build) that you're able to read them with both the new and the old code.